Swift/Sources/StateMachineMacros/Macros/StateMachineHashableMacro.swift

```diff
@@ -0,0 +1,93 @@
+//
+//  Copyright (c) 2024, Match Group, LLC
+//  BSD License, see LICENSE file for details
+//
+
+import SwiftSyntax
+import SwiftSyntaxMacros
+
+public struct StateMachineHashableMacro: ExtensionMacro {
+
+    public static func expansion(
+        of node: AttributeSyntax,
+        attachedTo declaration: some DeclGroupSyntax,
+        providingExtensionsOf type: some TypeSyntaxProtocol,
+        conformingTo protocols: [TypeSyntax],
+        in context: some MacroExpansionContext
+    ) throws -> [ExtensionDeclSyntax] {
+
+        guard let enumDecl: EnumDeclSyntax = declaration.as(EnumDeclSyntax.self)
+        else { throw StateMachineHashableMacroError.typeMustBeEnum }
+
+        let elements: [EnumCaseElementSyntax] = enumDecl
+            .memberBlock
+            .members
+            .compactMap { $0.as(MemberBlockItemSyntax.self) }
+            .map(\.decl)
+            .compactMap { $0.as(EnumCaseDeclSyntax.self) }
+            .flatMap(\.elements)
+
+        guard !elements.isEmpty
+        else { throw StateMachineHashableMacroError.enumMustHaveCases }
+
+        let enumCases: [String] = elements
+            .map(\.name.text)
+            .map { "case \($0)" }
+
+        let hashableIdentifierCases: [String] = elements
+            .map(\.name.text)
+            .map { "case .\($0):\nreturn .\($0)" }
+
+        var associatedValueCases: [String] = []
+        for element: EnumCaseElementSyntax in elements {
```

Comment:
curious why this is a `for...in` instead of `elements.forEach { }` since it doesn't seem to need to early exit the loop. Or even more safe, why not just:

```
let associatedValueCases: [String] = elements.map {
  // Existing logic here, but instead of "associatedValueCases.append(`case`)", just "return `case`"
}
```

Reply:
Done 👍